### PR TITLE
[UI/UX] Add customizable sysadmin info to page footer

### LINKED ISSUE #4375: Customizable sysadmin contact info on login page, home page, footer
**What problem are you trying to solve with Submitty**
When users (students, especially) experience a problem with a Submitty production installation, it should be obvious who to contact for help.

**Describe the way you'd like to solve this problem**
Let's collect some further information when we run CONFIGURE_SUBMITTY.sh, how about:
* email contact for system administration (if all problems should be sent to single place via simple email), OR,
* webpage url for a page (not hosted directly by Submitty) with detailed contact information (if figuring out who to report a problem to is nontrivial, or the process to report is involved)

Then, let's add a prominent 'help link' to the email or webpage on the 
* login page 
* home page (listing all courses) 

And also... 
* on the footer of every page (not sure about this one, the footer is already busy...   but it can be small!)

### PR DESCRIPTION
**Pull request by mukul-kmr-jha**
### Please check if the PR fulfills these requirements:

* [x] The PR title and message follows our [guidelines](https://submitty.org/developer/how_to_contribute)
* [ ] Documentation has been updated/added if relevant

### What is the current behavior?
When a student experiences some issues or problems within the Submitty, there is no mechanism for helping them out or providing them with a few helpful links.

### What is the new behavior?
Closes #4375 
Setting up Submitty now asks for the help link and admin's email address, and then provides these help link and an email on all the pages within Submitty.

### CODE REVIEW

File: site/app/controllers/GlobalController.php [hunk 1]
```diff
@@ -403,6 +403,14 @@ public function footer() {
                 }
             }
         }
+        // append the help links
+        if ($this->core->getConfig()->getSysAdminUrl() != '') {
+            $footer_links[] =  ["title" => "Report Issues", "url" => $this->core->getConfig()->getSysAdminUrl()];
+        }
+        if ($this->core->getConfig()->getSysAdminEmail() != '') {
+            $footer_links[] =  ["title" => "Email Sys. Admin", "url" => $this->core->getConfig()->getSysAdminEmail(), "isEmai" => true];
```
Comment (MasterOdin) [inline site/app/controllers/GlobalController.php]: typo here for `isEmai` (should be `isEmail`?)

Comment (mukul-kmr-jha) [reply site/app/controllers/GlobalController.php]: Done

**Comment by MasterOdin** [comment]
Sorry for the delay on this. There's some changes I'd like to suggest here, but that require a bit more major of a refactor of the CONFIGURE_SUBMITTY.py script on handling input. Once that's done, I'll properly review this, and get this merged, as it's more or less ready to go.

**Comment by bmcutler** [comment]
@mukul-kmr-jha 
please fix merge conflicts, thanks!

**Comment by mukul-kmr-jha** [comment]
Rebased, Please let me know the changes to be added... Thanks!

**Comment by MasterOdin** [comment]
So my thoughts on the matter are that:

1. These two values should not have a default value as I don't think we want students from other universities emailing us if their school does not customize things
1. We should make a capacity to "empty" the contents of a previously set field.

The latter is not something we support at all in the python script, and is what I'd like to add support for in separately from this PR, and then have this PR take advantage of that. My thoughts here would be to utilize typing "Ctrl+D" and catching the EOFError python naturally surfaces and use that to empty a field. I'd like to then add an additional attribute to the `get_input` field that designates whether or not something is mandatory or not (and if it should be possible to empty it). Let me know on slack if you'd like to take this on, and I can type up more clear thoughts on implementation details for you, otherwise it's something I hope to get to this week or next.

This PR though as it stands is more or less done and blocked by the addition of the above (which again, I'd like to happen separately from this PR just to make things a bit cleaner history wise).

File: site/tests/app/controllers/admin/ConfigurationControllerTester.php [hunk 2]
```diff
@@ -31,7 +31,7 @@ public function setUpConfig($seating_dirs = []): void {
...
+            'submitty' => '{"submitty_install_dir":' . json_encode($this->test_dir) . ',"submitty_repository":' . json_encode($this->test_dir) . ',"submitty_data_dir":' . json_encode($this->test_dir) . ',"autograding_log_path":' . json_encode($this->test_dir) . ',"site_log_path":' . json_encode($this->test_dir) . ',"submission_url":"http:\/\/localhost:1501","vcs_url":"","cgi_url":"http:\/\/localhost:1501\/cgi-bin","institution_name":"","username_change_text":"foo","institution_homepage":"" ,"sys_admin_email": "admin@example.com,"sys_admin_url": "https://example.com/admin","timezone":"America\/New_York","worker":false,"duck_special_effects" : false}',
```
Comment (MasterOdin) [inline site/tests/app/controllers/admin/ConfigurationControllerTester.php:34]: Need to escape `/` here: `https:\/\/example.com\/admin`

Comment (mukul-kmr-jha) [reply site/tests/app/controllers/admin/ConfigurationControllerTester.php:34]: > Yeah, the long strings were easy for convenience starting out, but we're clearly hitting the usefulness of that convenience over setting up an array and using `json_encode` over it. For now, I think just fixing the syntax, and I'll make a low priority issue to rewrite this to something a bit more sane.

Surely!

File: site/tests/app/controllers/admin/ConfigurationControllerTester.php [hunk 3]
```diff
@@ -31,7 +31,7 @@ public function setUpConfig($seating_dirs = []): void {
...
+            'submitty' => '{"submitty_install_dir":' . json_encode($this->test_dir) . ',"submitty_repository":' . json_encode($this->test_dir) . ',"submitty_data_dir":' . json_encode($this->test_dir) . ',"autograding_log_path":' . json_encode($this->test_dir) . ',"site_log_path":' . json_encode($this->test_dir) . ',"submission_url":"http:\/\/localhost:1501","vcs_url":"","cgi_url":"http:\/\/localhost:1501\/cgi-bin","institution_name":"","username_change_text":"foo","institution_homepage":"" ,"sys_admin_email": "admin@example.com,"sys_admin_url": "https:\/\/example.com\/admin","timezone":"America\/New_York","worker":false,"duck_special_effects" : false}',
```
Comment (MasterOdin) [inline site/tests/app/controllers/admin/ConfigurationControllerTester.php:34]: looks like there's a missing `"`

```suggestion
            'submitty' => '{"submitty_install_dir":' . json_encode($this->test_dir) . ',"submitty_repository":' . json_encode($this->test_dir) . ',"submitty_data_dir":' . json_encode($this->test_dir) . ',"autograding_log_path":' . json_encode($this->test_dir) . ',"site_log_path":' . json_encode($this->test_dir) . ',"submission_url":"http:\/\/localhost:1501","vcs_url":"","cgi_url":"http:\/\/localhost:1501\/cgi-bin","institution_name":"","username_change_text":"foo","institution_homepage":"" ,"sys_admin_email": "admin@example.com", "sys_admin_url": "https:\/\/example.com\/admin","timezone":"America\/New_York","worker":false,"duck_special_effects" : false}',
```

File: site/app/models/Config.php [hunk 4]
```diff
@@ -280,6 +296,9 @@ public function loadMasterConfigs($config_path) {
         $this->base_url = $submitty_json['submission_url'];
         $this->duck_banner_enabled = $submitty_json['duck_special_effects'] === true;
 
+        $this->sys_admin_email = $submitty_json['sys_admin_email'];
```
Comment (MasterOdin) [inline site/app/models/Config.php:299]:
```suggestion
        $this->sys_admin_email = $submitty_json['sys_admin_email'] ?? '';
```

File: site/app/models/Config.php [hunk 5]
```diff
@@ -280,6 +296,9 @@ public function loadMasterConfigs($config_path) {
...
+        $this->sys_admin_url = $submitty_json['sys_admin_url'];
```
Comment (MasterOdin) [inline site/app/models/Config.php:300]:
```suggestion
        $this->sys_admin_url = $submitty_json['sys_admin_url'] ?? '';
```

File: site/app/controllers/GlobalController.php [hunk 6]
```diff
@@ -407,6 +407,14 @@ public function footer() {
                 }
             }
         }
+        // append the help links
+        if ($this->core->getConfig()->getSysAdminUrl() !== '') {
+            $footer_links[] =  ["title" => "Report Issues", "url" => $this->core->getConfig()->getSysAdminUrl()];
+        }
+        if ($this->core->getConfig()->getSysAdminEmail() !== '') {
+            $footer_links[] =  ["title" => "Email Sys. Admin", "url" => $this->core->getConfig()->getSysAdminEmail(), "isEmail" => true];
```
Comment (MasterOdin) [inline site/app/controllers/GlobalController.php:415]:
```suggestion
            $footer_links[] =  ["title" => "Email Sys. Admin", "url" => $this->core->getConfig()->getSysAdminEmail(), "is_email" => true];
```

File: site/app/templates/GlobalFooter.twig [hunk 7]
```diff
@@ -14,7 +14,12 @@
                     {% if link['icon'] is defined %}
                         <i class="footer-link-icon fa fa-lg {{ link['icon'] }}"></i>
                     {% endif %}
-                    <a href="{{ link['url'] }}" target="_blank" class="black-btn">{{ link['title'] }}</a>
+                    {% if link['isEmail'] is defined %}
```
Comment (MasterOdin) [inline site/app/templates/GlobalFooter.twig:17]:
```suggestion
                    {% if link['is_email'] is defined %}
```

**Pull request merged**